packages/trip-details/src/TripDetails.story.tsx

```diff
@@ -0,0 +1,188 @@
+const bikeTransitBikeItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/bike-transit-bike.json");
+const eScooterRentalItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/e-scooter-rental.json");
+const eScooterRentalTransiteScooterRentalItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/e-scooter-transit-e-scooter.json");
+const parkAndRideItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/park-and-ride.json");
+const tncTransitTncItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/tnc-transit-tnc.json");
+const walkInterlinedTransitItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/walk-interlined-transit-walk.json");
+const walkOnlyItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/walk-only.json");
+const walkTransitWalkItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/walk-transit-walk.json");
+const walkTransitWalkTransitWalkItinerary = require("@opentripplanner/itinerary-body/src/__mocks__/itineraries/walk-transit-walk-transit-walk.json");
+
+const StyledTripDetails = styled(TripDetails)`
+  ${TripDetailsClasses.TripDetailsHeader} {
+    background-color: pink;
+  }
+`;
+
+const longDateFormat = "MMMM D, YYYY";
+
+// Custom slots for expandable detail sections.
+const CustomDepartureDetails = ({
+  departureDate
+}: DepartureDetailsProps): React.Element => (
+  <>
+    Custom messages about {departureDate.format(longDateFormat)} can be
+    constructed dynamically using any markup.
+  </>
+);
+
+const CustomFareDetails = ({
+  maxTNCFare,
+  minTNCFare,
+  transitFare
+}: FareDetailsProps): React.Element => (
+  <>
+    Custom details about fares (transitFare: {transitFare} (cents), minTNCFare:{" "}
+    {minTNCFare} and maxTNCFare: {maxTNCFare} can be constructed dynamically
+    using any markup.
+  </>
+);
+
+const CustomCaloriesDetails = ({
+  bikeSeconds,
+  calories,
+  walkSeconds
+}: CaloriesDetailsProps): React.Element => (
+  <>
+    Custom message about {calories} calories burned,
+    {walkSeconds} seconds and {bikeSeconds} seconds.
+  </>
+);
+
+// Wrapper component for TripDetails.
+const TripDetailsWrapper = ({
+  CaloriesDetails,
+  Component = TripDetails,
+  DepartureDetails,
+  FareDetails,
+  itinerary
+}: TripDetailsProps): React.Element => (
+  <Component
+    CaloriesDetails={CaloriesDetails}
+    currency={select("Currency", ["USD", "EUR"], "USD")}
+    DepartureDetails={DepartureDetails}
+    FareDetails={FareDetails}
+    itinerary={itinerary}
+  />
+);
+
+const intlDecorator = story => {
+  const useMessages = boolean("Use localized messages", true);
+  const locale = select("Locale", ["en-US", "fr"], "en-US");
+  const overrideDefaultMessages = boolean("Override default messages", false);
```

Comment:
The [@storybook/addon-knobs](https://www.npmjs.com/package/@storybook/addon-knobs) package is deprecated. Please use [Storybook args and argTypes](https://storybook.js.org/docs/react/writing-stories/args) instead.

Reply:
Updated in https://github.com/opentripplanner/otp-ui/pull/281/commits/da0eb14bbfb78531b95e966f84d82529b0c53db5.